scripts/release/prod/stage/snapshot.sh

```diff
@@ -1,13 +1,31 @@
 #!/usr/bin/env bash
 
+set -ex
+
+CHANNEL="$1"
+VERSION="$2"
+
+if [ -z "$CHANNEL" ] || [ -z "$VERSION" ]; then
+    echo "CHANNEL=$CHANNEL and VERSION=$VERSION variables must be defined."
+    exit 1
+fi
+
 echo
 date "+build_release begin SNAPSHOT stage %Y%m%d_%H%M%S"
 echo
 
-set -ex
+DEBS_DIR="$HOME/packages/deb/$CHANNEL"
+DEB="algorand_${CHANNEL}_linux-amd64_${VERSION}.deb"
+
+aws s3 cp "s3://algorand-builds/channel/$CHANNEL/$DEB" .
```

Comment:
A lot of the spots where you're using `aws` could use `updater` instead. Or `updater` could be extended to do this. The key benefit to this approach is that the path pattern is a function of updater, and therefore if we want to do a migration to another format all of the related logic is encapsulated in one place (`utils/s3/s3Helper.go`).

For this call, I think you would use `updater send -c $CHANNEL $DEB`. In some cases it uses environment variables to switch the upload context (i.e. configure an upload to go to `algorand-builds`, `algorand-internal`, etc), but you could also use `-b algorand-builds`. The travis web-configuration lets us set environment variables, and we configure the bucket for legacy scripts there (then developer machines use a different bucket for their builds).

Reply:
Good to know.  I'd rather not change this here, since this is the legacy pipeline and will quickly be replaced by `mule` tasks.  I'll look into this approach for those, however.

Reply:
Now that I think of it, @egieseke wrote a `mule` task that allows for the uploading and downloading of files, and perhaps copying as well.  I haven't looked at it yet but need to soon, so I'm not sure how this is being handled.  I would assume it's using the `boto` library, so perhaps this is moot?